src/sysutil.c

```diff
@@ -277,17 +277,10 @@ void DisplayExtendedSysHand(struct SysHandCursorProc * proc)
 {
     int i;
 
+    gPaletteBuffer[proc->pal_bank * 0x10  + 0x10E] =
 #if !NONMATCHING
-    u32 clk;
-    u16 * src, * dst, * _dst;
-
-    clk = GetGameClock();
-    dst = gPaletteBuffer;
-    _dst = dst + (proc->pal_bank * 0x10  + 0x10E);
-    src = &PAL_BUF_COLOR(Pal_08A1D448, gPlaySt.config.windowColor, (clk / 4) % 0x10);
-    *_dst = *src;
+        ((gPlaySt.config.windowColor << 4) + ((GetGameClock() / 4) % 0x10))[Pal_08A1D448];
```

Comment:
I think it would be reasonable to remove the `if !NONMATCHING` here, since this style of accessing the array is used elsewhere in the codebase (especially in the world map logic)